server/src/main/java/org/elasticsearch/http/HttpStats.java

```diff
@@ -49,14 +67,121 @@ public long getTotalOpen() {
         static final String HTTP = "http";
         static final String CURRENT_OPEN = "current_open";
         static final String TOTAL_OPENED = "total_opened";
+        static final String CLIENTS = "clients";
+        static final String CLIENT_ID = "id";
+        static final String CLIENT_AGENT = "agent";
+        static final String CLIENT_LOCAL_ADDRESS = "local_address";
+        static final String CLIENT_REMOTE_ADDRESS = "remote_address";
+        static final String CLIENT_LAST_URI = "last_uri";
+        static final String CLIENT_OPENED_TIME_MILLIS = "opened_time_millis";
+        static final String CLIENT_CLOSED_TIME_MILLIS = "closed_time_millis";
+        static final String CLIENT_LAST_REQUEST_TIME_MILLIS = "last_request_time_millis";
+        static final String CLIENT_REQUEST_COUNT = "request_count";
+        static final String CLIENT_REQUEST_SIZE_BYTES = "request_size_bytes";
+        static final String CLIENT_FORWARDED_FOR = "x_forwarded_for";
+        static final String CLIENT_OPAQUE_ID = "x_opaque_id";
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(Fields.HTTP);
         builder.field(Fields.CURRENT_OPEN, serverOpen);
         builder.field(Fields.TOTAL_OPENED, totalOpen);
+        builder.startArray(Fields.CLIENTS);
+        for (ClientStats clientStats : this.clientStats) {
+            builder.startObject();
+            builder.field(Fields.CLIENT_ID, clientStats.id);
+            builder.field(Fields.CLIENT_AGENT, clientStats.agent);
+            builder.field(Fields.CLIENT_LOCAL_ADDRESS, clientStats.localAddress);
+            builder.field(Fields.CLIENT_REMOTE_ADDRESS, clientStats.remoteAddress);
+            builder.field(Fields.CLIENT_LAST_URI, clientStats.lastUri);
+            builder.field(Fields.CLIENT_FORWARDED_FOR, clientStats.forwardedFor);
+            builder.field(Fields.CLIENT_OPAQUE_ID, clientStats.opaqueId);
```

Comment:
Should these string fields be protected with `if (Strings.hasText(clientStats.opaqueId)) {...}` so we don't output fields we don't care about? (or null check if we end up not using `Strings.EMPTY`)

Reply:
I think that makes sense for `forwardedFor` and `opaqueId`. An HTTP client will always have a remote address and at least one request URI.